pkg/inventory/inventory.go

```diff
@@ -106,7 +114,7 @@ func ValidateNoInventory(objs []*unstructured.Unstructured) error {
 // splitUnstructureds takes a slice of unstructured.Unstructured objects and
 // splits it into one slice that contains the inventory object templates and
 // another one that contains the remaining resources.
-func SplitUnstructureds(objs []*unstructured.Unstructured) (*unstructured.Unstructured, []*unstructured.Unstructured, error) {
+func SplitUnstructureds(factoryFunc UnstructuredToInvInfoFunc, objs []*unstructured.Unstructured) (InventoryInfo, []*unstructured.Unstructured, error) {
```

Comment:
This doesn't seem right. This function should only be needed in the cobra commands in this repo. I think we should just make this work only for ConfigMaps. Any users of the library that use a different resource type for inventory can handle this for their particular use-case.

Reply:
Since we need to split the list of resources to get the InventoryInfo, I added a function `InventoryInfo` in the provider interface for that purpose. Then the `SplitUnstructureds` is only used in the provider for ConfigMap.